lib/Sema/TypeCheckAvailability.cpp

```diff
@@ -2442,11 +2449,16 @@ void TypeChecker::diagnoseIfDeprecated(SourceRange ReferenceRange,
   // We match the behavior of clang to not report deprecation warnings
   // inside declarations that are themselves deprecated on all deployment
   // targets.
-  if (Where.isDeprecated()) {
+  if (Where.isDeprecated() || Where.ignoresDeprecation()) {
     return;
   }
 
   auto *ReferenceDC = Where.getDeclContext();
+  auto *moduleImportedFrom = ReferenceDC->findImportFor(ReferenceDC->getParentModule());
+  if (moduleImportedFrom.hasValue() && moduleImportedFrom.getValue().options.contains(ImportFlags::IgnoresDeprecationWarnings)) {
+    return;
+  }
```

Comment:
FWIW `Optional` has some syntactic sugar you could use here:

```suggestion
  if (moduleImportedFrom && moduleImportedFrom->options.contains(ImportFlags::IgnoresDeprecationWarnings)) {
    return;
  }
```